astropy/table/table.py

```diff
@@ -2373,32 +2373,32 @@ def __copy__(self):
         return self.copy(False)
 
     def __lt__(self, other):
-        if six.PY3:
-            return super(Table, self).__lt__(other)
-        elif six.PY2:
+        if six.PY2:
             raise TypeError("unorderable types: Table() < {0}".
                             format(str(type(other))))
+        else:
+            return super(Table, self).__lt__(other)
```

Comment:
Python3 allows using `super` without arguments. Again I'm not sure if this would fit the scope of this PR though. (there are several `super` calls guarded by `if not six.PY2`